app/api_alpha/validators.py

```diff
@@ -11,7 +11,12 @@ def ads_validate_rnbid(rnb_id):
 
 def bdg_is_active(rnb_id: str):
 
-    if Building.objects.filter(rnb_id=rnb_id, is_active=False).exists():
+    bdg = Building.objects.filter(rnb_id=rnb_id).first()
+
+    if bdg is None:
+        raise serializers.ValidationError(f'Building "{rnb_id}" does not exist.')
+
+    if not bdg.is_active:
```

Comment:
```suggestion
    bdg = Building.objects.get(rnb_id=rnb_id)

    if not bdg.is_active:
```

(j'ai pas testé)
Normalement on a qu'une seule ligne dans la table building avec un RNB ID donné. [get](https://docs.djangoproject.com/en/5.1/topics/db/queries/#retrieving-a-single-object-with-get) semble adapté et va lever une exception si on en trouve plusieurs ou aucun.

En fait en disant `Building.objects.filter(rnb_id=rnb_id, is_active=False).exists()`on force la base à tout scanner, parce que même quand il trouve la ligne avec le rnb_id spécifié et que le building est actif, il DOIT scanner toute la table au cas ou une autre ligne existe ailleurs avec rnb_id et is_active=False. Alors que nous on sait que ça ne peut pas arriver (rnb_id unique), donc je pense que ta nouvelle manière de faire est non seulement plus simple à mettre en place qu'un index, mais surtout la bonne solution.